### PR TITLE
CXX-699 compute nextBatchSize() correctly

### PR DESCRIPTION
**Pull request by amidvidy**
need to write a test.

### CODE REVIEW

**Comment by amidvidy** [comment]
added test.

**Comment by markbenvenuto** [comment]
:+1:

File: src/mongo/client/dbclientcursor.cpp [hunk 1]
```diff
@@ -92,16 +92,11 @@ void DBClientCursor::_finishConsInit() {
 }
 
 int DBClientCursor::nextBatchSize() {
```
Comment (acmorrow) [inline src/mongo/client/dbclientcursor.cpp]: Is there some documentation somewhere that specifies that this is the correct interpretation? This has been a potent source of errors in the past

File: src/mongo/integration/standalone/dbclient_test.cpp [hunk 2]
```diff
@@ -1413,5 +1413,60 @@ TEST_F(DBClientTest, ConnectionStringWithTestDB) {
...
+                                 << "c"
+                                 << "dddddddddddddddddddddddddddd"));
+        }
+        bulk.execute(&WriteConcern::acknowledged, &result);
+        ASSERT_EQUALS(static_cast<std::size_t>(result.nInserted()), bulkSize);
+    }
+
+    {
+        // A sanity check.
+        auto_ptr<DBClientCursor> limitQuery1 = c.query(TEST_NS, mongo::Query("{}"), 1);
```
Comment (acmorrow) [inline src/mongo/integration/standalone/dbclient_test.cpp]: I see that this test changes the limit, but I think it is sticking with the default batch size? Is there a way we can explore extremal values of batchSize and ensure that things still work?

**Comment by acmorrow** [comment]
LGTM mod two questions

File: src/mongo/client/dbclientcursor.cpp (hunk 1 above)
Comment (amidvidy) [inline src/mongo/client/dbclientcursor.cpp]: this is the same logic in the server

**Comment by amidvidy** [comment]
added additional tests w/ non-default batchSize

**Comment by acmorrow** [comment]
LGTM

**Pull request merged**